### DISCUSSION TITLE
Using Milvus to get the most dis-similar image as the first output

### DISCUSSION THREAD

**Discussion by ZubeenK** [milvus-io/milvus#24104] [Q&A and General discussion]
Hi, I have a use case, where I need to find the most dis-similar image as the first output and then the second most dis-similar image and so on (based on Euclidean distance). One way to do this can be to return all the images using top_k and then reverse the list, but I don't think this as an efficient solution, Is there a better way to do this?

**Comment by yhmo** [comment]
The major purpose of ANN search is to find out the most similar embeddings. I think currently there is no better way to meet your requirement. 
Note: Milvus topk limit is 16384. If your embeddings count exceeds 16384, you need to split data into partitions or collections, and call search() for each collection/partition, pick the dis-similar embeddings.

**Comment by xiaofan-luan** [comment]
> Hi, I have a use case, where I need to find the most dis-similar image as the first output and then the second most dis-similar image and so on (based on Euclidean distance). One way to do this can be to return all the images using top_k and then reverse the list, but I don't think this as an efficient solution, Is there a better way to do this?

Is there any index or paper support most dis-similar search?
